src/main/scala/variant/Horde.scala

```diff
@@ -43,7 +43,10 @@ case object Horde
   override def valid(situation: Situation, strict: Boolean) =
     situation.board.kingOf(White).isEmpty && validSide(situation.board, strict)(
       Black
-    ) && !pawnsOnPromotionRank(situation.board, White)
+    ) && !pawnsOnPromotionRank(situation.board, White) && hasValidCheckers(strict, situation)
+
+  override def hasValidCheckers(strict: Boolean, situation: Situation): Boolean =
```

Comment:
We only need check for Black, because White has no King, so we can optimize a bit by ignoring White.